servicetalk-http-netty/src/test/java/io/servicetalk/http/netty/ProxyTunnel.java

```diff
@@ -105,72 +106,65 @@ int connectCount() {
+        return bos.toString(UTF_8.name());
     }
 
-    private void handleRequest(final Socket socket, final InputStream in, final String initialLine) throws IOException,
-            ExecutionException, InterruptedException {
-        try {
-            if (initialLine.startsWith(CONNECT_PREFIX)) {
-                final int end = initialLine.indexOf(' ', CONNECT_PREFIX.length());
-                final String authority = initialLine.substring(CONNECT_PREFIX.length(), end);
-                final String protocol = initialLine.substring(end + 1);
-                final int colon = authority.indexOf(':');
-                final String host = authority.substring(0, colon);
-                final int port = Integer.parseInt(authority.substring(colon + 1));
-
-                try (Socket clientSocket = new Socket(host, port)) {
-                    connectCount.incrementAndGet();
-                    final OutputStream out = socket.getOutputStream();
-                    out.write((protocol + " 200 Connection established\r\n\r\n").getBytes(UTF_8));
-                    out.flush();
-
-                    final InputStream cin = clientSocket.getInputStream();
-                    Future<Void> f = executor.submit(() -> {
-                        copyStream(out, cin);
-                        return null;
-                    });
-                    copyStream(clientSocket.getOutputStream(), in);
-                    f.get(); // wait for the copy of proxy client input to server output to finish copying.
-                }
-            } else {
-                throw new RuntimeException("Unrecognized initial line: " + initialLine);
+    private void handleRequest(final Socket socket, final String initialLine) throws IOException {
+        if (initialLine.startsWith(CONNECT_PREFIX)) {
+            final int end = initialLine.indexOf(' ', CONNECT_PREFIX.length());
+            final String authority = initialLine.substring(CONNECT_PREFIX.length(), end);
+            final String protocol = initialLine.substring(end + 1);
+            final int colon = authority.indexOf(':');
+            final String host = authority.substring(0, colon);
+            final int port = Integer.parseInt(authority.substring(colon + 1));
+
+            try (Socket clientSocket = new Socket(host, port)) {
+                connectCount.incrementAndGet();
+                final OutputStream out = socket.getOutputStream();
+                out.write((protocol + " 200 Connection established\r\n\r\n").getBytes(UTF_8));
+                out.flush();
+
+                executor.submit(() -> {
+                    try {
+                        copyStream(out, clientSocket.getInputStream());
+                    } finally {
+                        // We are simulating a proxy that doesn't do half closure. The proxy should close the server
+                        // socket as soon as the server read is done.
+                        socket.close();
```

Comment:
1. Thanks for leaving a comment here. We can even make it more visible with something like:
```
// ! NEVER CHANGE:
``` 
😄 

Consider also referencing which tests require this behavior: `GracefulConnectionClosureHandlingTest` and `ConnectionCloseHeaderHandlingTest`.

2. Do you mind reverting to how the closure worked before #1816? Because tests verify graceful closure triggered by both sides, we also need to close if client sends FIN. Better to keep closure logic inside `copyStream(...)`.

Reply:
I think it is more clear to close the socket instead of each individual stream (which maybe confused to mean half-closure is desired). The client and server socket will always be closed at the conclusion of copying (either in try/finally for `clientSocket`, or in a `finally` outside the `handleRequest` method).